NearestGene.pm

```diff
@@ -74,12 +80,21 @@ sub new {
   
   my $params = $self->params;
 
+  # get output format
+  $char_sep = ":" if ($self->{config}->{output_format} eq 'vcf');
+
   foreach my $param(@$params) {
     my ($key, $val) = split('=', $param);
-    die("ERROR: Failed to parse parameter $param\n") unless defined($key) && defined($val);
-    $CONFIG{$key} = $val;
+
+    if($key eq "both_directions") {
+      $self->{both_directions} = 1;
+    }
+    else {
+      die("ERROR: Failed to parse parameter $param\n") unless defined($key) && defined($val);
+      $CONFIG{$key} = $val;
+    }
```

Comment:
```suggestion
    die("ERROR: Failed to parse parameter $param\n") unless defined($key) && defined($val);
    if($key eq "both_directions") {
      $self->{both_directions} = 1;
    }
    else {
      $CONFIG{$key} = $val;
    }
```

`die` should come before if-else statement. We should check the key exist before checking its value.